app/models/concerns/patient_date_validator.rb

```diff
@@ -6,15 +6,20 @@ class PatientDateValidator < ActiveModel::Validator
 
   def validate(record)
     year_start = Date.new(2020, 1, 1)
-    month_ahead = 30.days.from_now.to_date
-    month_behind = 30.days.ago.to_date
-    validate_between_dates(record, :date_of_birth, Date.new(1900, 1, 1), Time.now.to_date) if record.date_of_birth_changed?
-    validate_between_dates(record, :last_date_of_exposure, year_start, month_ahead) if record.last_date_of_exposure_changed?
-    validate_between_dates(record, :symptom_onset, year_start, month_ahead) if record.symptom_onset_changed?
-    validate_between_dates(record, :extended_isolation, month_behind, month_ahead) if record.extended_isolation_changed?
-    validate_between_dates(record, :date_of_departure, year_start, month_ahead) if record.date_of_departure_changed?
-    validate_between_dates(record, :date_of_arrival, year_start, month_ahead) if record.date_of_arrival_changed?
-    validate_between_dates(record, :additional_planned_travel_start_date, year_start, month_ahead) if record.additional_planned_travel_start_date_changed?
-    validate_between_dates(record, :additional_planned_travel_end_date, year_start, month_ahead) if record.additional_planned_travel_end_date_changed?
+    # NOTE: Commented out for rubocop until other validations are uncommented
+    # month_ahead = 30.days.from_now.to_date
+    # month_behind = 30.days.ago.to_date
+    # This is to handle the edge case when certain time zones such as Samoa are on the new day before UTC
+    earliest_tz_today = Time.now.in_time_zone(ActiveSupport::TimeZone.all.last).to_date
+    validate_between_dates(record, :symptom_onset, year_start, earliest_tz_today) if record.symptom_onset_changed?
```

Comment:
Since this check assumes the date could be based on any timezone and the front end validation is based on the timezone of the user's system, there is an edge case that's a little weird. As soon as the earliest timezone is in the next day, you can import a symptom onset dated the next day, but then if you edit the monitoree case information in the UI, it will fail the front end validation, forcing you to change the SOD in order to change any other case information (8/3 is today):

![image](https://user-images.githubusercontent.com/8235974/128092496-0bf3be6f-00e5-4fda-bff5-d8d02ba89cb0.png)

This is definitely an edge case, but I _could_ potentially see it happening. I guess ideally we'd align them so the front end validation is also based on the earliest timezone. Thoughts?

Reply:
Yeah that definitely is a possibility. One potential solution I'm thinking of would be to somehow get the timezone info of the client upon import for dynamic validation but I'm thinking that might be a bit complicated and overkill. If a user somehow does end up in this edge case though, I also don't think it would be necessarily wrong for them to be forced to change SO date to today.

Reply:
Yeah I guess I'm ok with that since we can expect the user's system to almost always be in the same timezone as the monitoree